contracts/staking/BaseShareField.sol

```diff
@@ -48,10 +83,22 @@ contract BaseShareField {
     function _audit(address user) internal virtual {
         UserInfo storage userInfo = users[user];
         if (userInfo.amount > 0) {
+            uint256 blocksPaid = userInfo.lastRewardTime.sub(userInfo.multiplierResetTime);
+            uint256 blocksPassedFirstMonth = Math.min(172800,block.number.sub(userInfo.multiplierResetTime)); //172800 is equivalent of month in blocks 
+            uint256 blocksToPay = block.number.sub(userInfo.lastRewardTime);
+            uint256 firstMonthBlocksToPay = blocksPaid >= (172800) ? 0 : blocksPassedFirstMonth.sub(blocksPaid);
+            uint256 fullBlocksToPay = blocksToPay.sub(firstMonthBlocksToPay);
+            
             uint pending = userInfo.amount.mul(accAmountPerShare).div(1e12).sub(userInfo.rewardDebt);
+           
+            if (blocksToPay != 0){
+                uint rewardPerBlock = pending.mul(uint256(10 ** 12)).div(blocksToPay).div(uint256(10 ** 12)); // increase resolution
+                pending  = ((firstMonthBlocksToPay.mul(50*10**18).div(100)) + fullBlocksToPay).mul(rewardPerBlock).div(1e18); // divide 1e16 so reduce it to 18decimals
```

Comment:
comment says 1e16 but you divide 1e18
better comments required on math

Reply:
explain the calculation you are doing,
even maybe add in thee function comments the formulas from the word file